Meadow/Creatures/ScavengerController.cs

```diff
@@ -32,12 +32,12 @@ private static void ScavengerHand_Update(ILContext il)
                 for (int i = 0; i < 2; i++)
                 {
                     c.GotoNext(MoveType.After,
-                        i => i.MatchLdarg(0),
-                        i => i.MatchCall<ScavengerGraphics.ScavengerHand>("get_scavenger"),
-                        i => i.MatchLdfld<UpdatableAndDeletable>("room"),
-                        i => i.MatchLdloc(out _),
+                        i => i.MatchCallvirt<Creature>("get_mainBodyChunk"),
+                        i => i.MatchLdflda<BodyChunk>("pos"),
+                        i => i.MatchLdfld<UnityEngine.Vector2>("y"),
+                        i => i.MatchNewobj<UnityEngine.Vector2>(),
                         i => i.MatchCallOrCallvirt<Room>("FloatWaterLevel")
```

Comment:
Never use strings when IL Hooking. This mistake is in the original and is why the hook soft broke instead of not compiling.

```suggestion
                        i => i.MatchCallvirt<Creature>(typeof(Creature).GetProperty(nameof(Creature.mainBodyChunk)).GetGetMethod()),
                        i => i.MatchLdflda<BodyChunk>(nameof(BodyChunk.pos)),
                        i => i.MatchLdfld<Vector2>(nameof(Vector2.y)),
                        i => i.MatchNewobj<Vector2>(),
                        i => i.MatchCallOrCallvirt<Room>(nameof(Room.FloatWaterLevel))
```